src/annoymodule.cc

```diff
@@ -236,30 +236,37 @@ py_an_save(py_annoy *self, PyObject *args, PyObject *kwargs) {
 
 PyObject*
 get_nns_to_python(const vector<int32_t>& result, const vector<float>& distances, int include_distances) {
-  PyObject* l = PyList_New(result.size());
+  PyObject* l = NULL;
+  PyObject* d = NULL;
+  PyObject* t = NULL;
+  l = PyList_New(result.size());
   if (l == NULL) {
-    return NULL;
+    goto error;
   }
   for (size_t i = 0; i < result.size(); i++)
     PyList_SetItem(l, i, PyInt_FromLong(result[i]));
```

Comment:
Not sure what's going on with `PyInt` vs. `PyLong`, but this can also fail: https://docs.python.org/3/c-api/long.html#c.PyLong_FromLong

`PyList_SetItem` returns -1 on failure, but I believe that's only if the index is out of bounds, which should never happen.